File: MyCarApp/MyCarApp/Car.swift
import Foundation

struct Car: Identifiable, Codable {
    let id: UUID
    let make: String
    let model: String
    let bidCarsMake: String
    let bidCarsModel: String
    let avByMake: String
    let avByModel: String
    let followData: [FollowData]
}

struct FollowData: Codable {
    let userId: UUID
    let runStatus: String
    let yearFrom: Int
    let yearTo: Int
    let driveType: Int
    let engineCapacity: Int
}

File: car-request-service/src/main/kotlin/by/judoka/car_request_service/model/Car.kt
package by.judoka.car_request_service.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.util.UUID
import kotlin.collections.ArrayList

@Document("car-request")
class Car(
    @Id var id: UUID = UUID.randomUUID(),

    var make: String = "",
    var model: String = "",

    var bidCarsMake: String = "",
    var bidCarsModel: String = "",

    var avByMake: String = "",
    var avByModel: String = "",

    var followData: List<FollowCarData> = ArrayList()
)
